dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtilTest.java

```diff
@@ -880,6 +888,148 @@ public void testOnePullRequestForMultipleDockerfilesInSameRepo() throws Exceptio
     }
 
+    @Test
+    public void testPullRequestBlockNotExecutedWhenDockerfileIsNotModified() throws Exception {
+        Map<String, Object> nsMap = ImmutableMap.of(Constants.IMG,
+                "image", Constants.TAG,
+                "tag", Constants.STORE,
+                "store");
+        Namespace ns = new Namespace(nsMap);
+        GitForkBranch gitForkBranch = new GitForkBranch("image", "tag", null, "");
+        Multimap<String, GitHubContentToProcess> pathToDockerfilesInParentRepo = HashMultimap.create();
+        pathToDockerfilesInParentRepo.put("repo", new GitHubContentToProcess(null, null, "df11"));
+
+        GHRepository forkedRepo = mock(GHRepository.class);
+        when(forkedRepo.isFork()).thenReturn(true);
+        when(forkedRepo.getFullName()).thenReturn("forkedrepo");
+
+        GHRepository parentRepo = mock(GHRepository.class);
+        when(parentRepo.getFullName()).thenReturn("repo");
+        when(forkedRepo.getParent()).thenReturn(parentRepo);
+
+        //GHRepository parent = mock(GHRepository.class);
+        String defaultBranch = "default";
+        when(parentRepo.getDefaultBranch()).thenReturn(defaultBranch);
+        GHBranch parentBranch = mock(GHBranch.class);
+        String sha = "abcdef";
+        when(parentBranch.getSHA1()).thenReturn(sha);
+        when(parentRepo.getBranch(defaultBranch)).thenReturn(parentBranch);
+
+        gitHubUtil = mock(GitHubUtil.class);
+        dockerfileGitHubUtil = Mockito.spy(new DockerfileGitHubUtil(gitHubUtil));
+        InputStream stream = new ByteArrayInputStream("someContent".getBytes(StandardCharsets.UTF_8));
+        GHContent forkedRepoContent = mock(GHContent.class);
+        when(forkedRepoContent.read()).thenReturn(stream);
+
+        RateLimiter rateLimiter = mock(RateLimiter.class);
+        when(gitHubUtil.tryRetrievingContent(eq(forkedRepo),
+                eq("df11"), eq("image-tag"))).thenReturn(forkedRepoContent);
+
+        //Dockerfile not modified
+        doReturn(false).when(dockerfileGitHubUtil).modifyOnGithub(eq(forkedRepoContent),
+                eq("image-tag"),
+                eq("image"),
+                eq("tag"),
+                eq(null),
+                eq(null));
+
+        doNothing().when(rateLimiter).consume();
+
+        dockerfileGitHubUtil.changeDockerfiles(ns, pathToDockerfilesInParentRepo,
+                new GitHubContentToProcess(forkedRepo, parentRepo, ""), new ArrayList<>(),
+                gitForkBranch, rateLimiter);
+
+        // Dockerfile retrieved from the repo
+        verify(dockerfileGitHubUtil).tryRetrievingContent(eq(forkedRepo),
+                eq("df11"), eq("image-tag"));
+
+        // Trying to modify Dockerfile
+        verify(dockerfileGitHubUtil).modifyOnGithub(any(), eq("image-tag"), eq("image"), eq("tag"), any(), any());
+
+        // PR creation block not executed
+        verify(dockerfileGitHubUtil, times(0)).createPullReq(eq(parentRepo),
```

Comment:
Prefer `never()` to `times(0)`, https://stackoverflow.com/questions/12862659/how-to-verify-that-a-specific-method-was-not-called-using-mockito